### DISCUSSION TITLE
game crashes when trying to play any sound on windows

### DISCUSSION THREAD

**Discussion by lol4meh** [HaxeFlixel/flixel#2848] [Q&A]
today i was planning to add some sound effects to my game but everytime i used FlxG.sound.play() or FlxG.sound.playMusic() the game would crash when trying to execute any sound related lines. weird thing is, this happened only on windows and when i tried playing the sound effect when i compiled the build in html5 it worked completely fine. even the snippet for FlxSound crashed when built on windows, but worked completely fine on html5. dunno how to fix this, would be greatly appreaciated to know the issue. here is the snippet i used: 

```hx
package;

import flixel.FlxG;
import flixel.FlxSprite;
import flixel.FlxState;
import flixel.math.FlxMath;
import flixel.text.FlxText;
import flixel.ui.FlxButton;

class PlayState extends FlxState
{
	override public function create()
	{
               super.create();

		var soundtest:FlxButton = new FlxButton(0, 0, "play jumpscare", playjumpscaresound);
		soundtest.screenCenter();
		add(soundtest);
	}

	function playjumpscaresound()
	{
		FlxG.sound.play("assets/jumpscare.ogg");
	}
}
```

to note: when opening it in a debug windows build, it gives the error "uncaught exception: null access" after crashing. hopefully this helps.

**Comment by Geokureli** [comment]
In your project xml see if you have an asset tag that omits .ogg files on specific targets

**Reply by lol4meh** [reply]
i don't see any tags that would omit .ogg files in my project.xml, in fact i have not changed anything in it since i have created the template using flixel tpl. tried playing .mp3 sound files aswell, but those crash too.

**Reply by Geokureli** [reply]
can you share the project.xml just to be sure. Also, find the sound file in vscode's file explorer, right click it, select "copy relative file path" and paste it here and see if it's different from your "assets/jumpscare.ogg" path, at all

**Reply by lol4meh** [reply]
here is my project.xml:

```
<?xml version="1.0" encoding="utf-8"?>
<project xmlns="http://lime.software/project/1.0.2" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xsi:schemaLocation="http://lime.software/project/1.0.2 http://lime.software/xsd/project-1.0.2.xsd">

	<!-- _________________________ Application Settings _________________________ -->

	<app title="test" file="test" main="Main" version="0.0.1" company="HaxeFlixel" />

	<!--The flixel preloader is not accurate in Chrome. You can use it regularly if you embed the swf into a html file
		or you can set the actual size of your file manually at "FlxPreloaderBase-onUpdate-bytesTotal"-->
	<app preloader="flixel.system.FlxPreloader" />

	<!--Minimum without FLX_NO_GAMEPAD: 11.8, without FLX_NO_NATIVE_CURSOR: 11.2-->
	<set name="SWF_VERSION" value="11.8" />

	<!-- ____________________________ Window Settings ___________________________ -->

	<!--These window settings apply to all targets-->
	<window width="640" height="480" fps="60" background="#000000" hardware="true" vsync="false" />

	<!--HTML5-specific-->
	<window if="html5" resizable="false" />

	<!--Desktop-specific-->
	<window if="desktop" orientation="landscape" fullscreen="false" resizable="true" />

	<!--Mobile-specific-->
	<window if="mobile" orientation="landscape" fullscreen="true" width="0" height="0" />

	<!-- _____________________________ Path Settings ____________________________ -->

	<set name="BUILD_DIR" value="export" />
	<source path="source" />
	<assets path="assets" />

	<!-- _______________________________ Libraries ______________________________ -->

	<haxelib name="flixel" />

	<!--In case you want to use the addons package-->
	<!--<haxelib name="flixel-addons" />-->

	<!--In case you want to use the ui package-->
	<!--<haxelib name="flixel-ui" />-->

	<!--In case you want to use nape with flixel-->
	<!--<haxelib name="nape-haxe4" />-->

	<!-- ______________________________ Haxedefines _____________________________ -->

	<!--Enable the Flixel core recording system-->
	<!--<haxedef name="FLX_RECORD" />-->

	<!--Disable the right and middle mouse buttons-->
	<!--<haxedef name="FLX_NO_MOUSE_ADVANCED" />-->

	<!--Disable the native cursor API on Flash-->
	<!--<haxedef name="FLX_NO_NATIVE_CURSOR" />-->

	<!--Optimise inputs, be careful you will get null errors if you don't use conditionals in your game-->
	<haxedef name="FLX_NO_MOUSE" if="mobile" />
	<haxedef name="FLX_NO_KEYBOARD" if="mobile" />
	<haxedef name="FLX_NO_TOUCH" if="desktop" />
	<!--<haxedef name="FLX_NO_GAMEPAD" />-->

	<!--Disable the Flixel core sound tray-->
	<!--<haxedef name="FLX_NO_SOUND_TRAY" />-->

	<!--Disable the Flixel sound management code-->
	<!--<haxedef name="FLX_NO_SOUND_SYSTEM" />-->

	<!--Disable the Flixel core focus lost screen-->
	<!--<haxedef name="FLX_NO_FOCUS_LOST_SCREEN" />-->

	<!--Disable the Flixel core debugger. Automatically gets set whenever you compile in release mode!-->
	<haxedef name="FLX_NO_DEBUG" unless="debug" />

	<!--Enable this for Nape release builds for a serious peformance improvement-->
	<haxedef name="NAPE_RELEASE_BUILD" unless="debug" />

	<!-- _________________________________ Custom _______________________________ -->

	<!--Place custom nodes like icons here (higher priority to override the HaxeFlixel icon)-->
</project>
```

when i right click on the file and select "copy relative path", it pastes this in my clipboard (i opened my "projects" folder in vscode which contains the "test" template): test\assets\jumpscare.ogg

changed the FlxG.sound.play() with the path above (so it would be FlxG.sound.play("test/assets/jumpscare.ogg"), but this still results in the game crashing whenever i click on the button. i also tried it with different audio files, like FlxG.sound.play("assets/testmusic.mp3"), but it seems like there isnt a problem with the audio file.

**Reply by lol4meh** [reply]
here is the error it gives after crashing on neko:
![error](https://github.com/HaxeFlixel/flixel/assets/137432401/27feef7f-1020-4c01-88ac-d495d679b558)

**Reply by Geokureli** [reply]
test/assets/ is weird. is your project.xml inside the test folder as well? why isn't your vscode using the "test" folder as the project folder?

Can you share the project by making a public github repo? I'd have more luck finding the cause if I could run it myself

**Reply by lol4meh** [reply]
made a public repo as you said. i really dont know if i have an issue with the code

here is the link to the repo: https://github.com/lol4meh/flixelsoundtest

**Reply by Geokureli** [reply]
not seeing anything obvious, I'll take a closer look on monday, when I can

**Reply by lol4meh** [reply]
alright, thank you for helping :)

**Reply by lol4meh** [reply]
its been quite a while now, and i have mainly forgotten about this discussion, but just to ask: have you found the source of the problem? i tried it again as im writing this and the sound crashing problem is still occuring.

**Reply by Geokureli** [reply]
I tried your project, sorry for the delay. it doesn't crash for me, but the sound doesn't play, for some reason openfl finds that the asset exists and tries to load it, but fails to do so. I tried opening your sound in audacity, saving it as a new file and trying that, it also didn't work. I grabbed a sound from another project and tried that, it worked. I have no idea what's going on with that sound but lime says it exists but `NativeCFFI.lime_audio_load_file(path, audioBuffer)` returns null. This may be why it's crashing for you

**Reply by Geokureli** [reply]
Passing the buck onto lime since this isn't a flixel problem. For a short term fix I think you should use a different sound since the one your using seems corrupted or something.

https://github.com/openfl/lime/issues/1704
